src/xunit.performance.api/XunitPerformanceHarnessOptions.cs

```diff
@@ -44,6 +45,31 @@ public string OutputDirectory
             }
         }
 
+        [Option("perf:tmpdir", Required = false, HelpText = "Specifies a writable directory to storage temporal files.")]
+        public string TmpDirectory
+        {
+            get { return _tmpDirectory; }
+
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    throw new Exception("The temporal directory name cannot be null, empty or white space.");
+                }
+
+                if (value.Any(c => Path.GetInvalidPathChars().Contains(c)))
+                {
+                    throw new Exception("Specified temporal directory name contains invalid path characters.");
+                }
+
+                _tmpDirectory = Path.IsPathRooted(value) ? value : Path.GetFullPath(value);
+                if (!Directory.Exists(_tmpDirectory))
```

Comment:
We should make sure that this fails gracefully and it's clear to the user what happened if this fails (e.g. they don't have permissions).